### ISSUE TITLE
[sort-comp] allow configuring the order of `props` relative to `state`

### ISSUE THREAD

**Issue by pelotom** [yannickcr/eslint-plugin-react#892]
Currently the `react/sort-comp` rule disallows putting `props` before `state`:

``` javascript
class C extends Component {
  props = {} // warning: props should be placed after state
  state = {}
}
```

It seems from the docs that there's no way to configure this short of turning off the rule completely. Really I think the default should be the other way around; since `props` represent the public API of a component, I want them to be the very first thing I see. But in any case it should be configurable.

**Comment by ljharb** [comment]
From reading the docs, it seems like class properties (a stage 2 proposal) simply don't have a way to configure them, unless they fall in "everything else".

I agree that "class properties" should be its own group, and that by default, props should go before state, which should go before other class properties.

**Comment by billschaller** [comment]
Yeah, this rule seems to fail hard at handling either class properties or flow-annotated property declarations.

**Comment by ljharb** [comment]
`sort-comp` now has `instance-methods` and `instance-variables` as available groups.
